src/components/editor/Bindings/SchemaInference/Button.tsx

```diff
@@ -53,36 +53,34 @@ function SchemaInferenceButton() {
             )
                 .then(
                     (response) => {
+                        let inferredSchema = null;
                         if (Object.hasOwn(collectionData.spec, 'writeSchema')) {
-                            const { writeSchema, ...additionalSpecKeys } =
+                            const { ...additionalSpecKeys } =
                                 collectionData.spec;
 
-                            setInferredSpec(
-                                !isEmpty(response.schema)
-                                    ? {
-                                          writeSchema:
-                                              collectionData.spec.writeSchema,
-                                          readSchema: response.schema,
-                                          ...additionalSpecKeys,
-                                      }
-                                    : null
-                            );
+                            inferredSchema = !isEmpty(response.schema)
+                                ? {
+                                      ...additionalSpecKeys,
+                                      writeSchema:
+                                          collectionData.spec.writeSchema,
+                                      readSchema: response.schema,
+                                  }
```

Comment:
This is the main fix. Need to put the spread operator at the top so that the `writeSchema` and `readSchema` are not overwritten by ones that could be stored in the `collectionData.spec`